lib/net.js

```diff
@@ -267,9 +267,9 @@ function writeAfterFIN(chunk, encoding, cb) {
 
   var er = new Error('This socket has been ended by the other party');
   er.code = 'EPIPE';
-  var self = this;
+
   // TODO: defer error events consistently everywhere, not just the cb
-  self.emit('error', er);
+  this.emit('error', er);
```

Comment:
This is borderline outside the scope of this commit. It seemed silly to reassign `this` to `self` for just this one line, so why not.